editors/code/src/main.ts

```diff
@@ -67,14 +47,52 @@ async function tryActivate(context: vscode.ExtensionContext) {
 
     const workspaceFolder = vscode.workspace.workspaceFolders?.[0];
     if (workspaceFolder === undefined) {
-        throw new Error("no folder is opened");
+        const rustDocuments = vscode.workspace.textDocuments.filter(document => isRustDocument(document));
+        if (rustDocuments.length > 0) {
+            ctx = await Ctx.create(config, context, serverPath, { kind: 'Detached Files', files: rustDocuments });
+        } else {
+            throw new Error("no rust files are opened");
```

Comment:
Oh, didn't realize we still have this `vscode.workspace.workspaceFolders?.[0]` logic on the client side. It's wrong and should be removed (unrelated to this PR of course)

Reply:
Time to break the client again, I guess.

What should be the alternative?
`const target = vscode.workspace.workspaceFolders![0]; // safe, see main activate()`
?

Reply:
As usual, just don't do it at all :-)

We use workspace folder for two things:

* to set cwd for the server
* in task provider

The serever is not relying on cwd (we are careful with using AbsPaths everywhere), so, presumably, that logic can be removed. 

The tasks should be fixed to collect the task from all workspace folders, and not just from the first one.

Reply:
> to set cwd for the server

I feel that could be connected with the debug.ts logic (it's a 3rd place where we use `vscode.workspace.workspaceFolders`), but I'll give it a try and check how it goes.

Not doing anything in this PR, just in case.